pkg/s3store/s3store.go

```diff
@@ -100,6 +101,91 @@ import (
 // considered valid into a header value according to RFC2616.
 var nonPrintableRegexp = regexp.MustCompile(`[^\x09\x20-\x7E]`)
 
+var _ handler.ServerDataStore = &S3Store{}
+
+func (store S3Store) AsServableUpload(upload handler.Upload) handler.ServableUpload {
+	return &S3ServableUpload{
+		store:  &store,
+		upload: upload.(*s3Upload),
+	}
+}
+
+type S3ServableUpload struct {
+	store  *S3Store
+	upload *s3Upload
+}
+
+func (su *S3ServableUpload) ServeContent(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
```

Comment:
Before merging, we should move these functions down in the file as they are not the most important ones. Maybe place the logic next to the other optional interfaces that S3Store implements.

Reply:
:+1: